### DISCUSSION TITLE
3rd Party Project Naming / Trademarks

### DISCUSSION THREAD

**Discussion by mackenly** [Kareadita/Kavita#2949] [Ideas]
### Idea Description

Hello, I've built a tool for dropping/selecting ebook files and organizing them into Kavita's folder structure. This is a pain point I've experienced when importing several books at once (it becomes a rather tedious task to create all the folders). It's currently still private, but named "Kavita Import Tool." Before I make it public (open source under GPL) are there concerns around using Kavita in the name? Thank you for an awesome tool for managing ebook libraries!

### Idea Category

User Experience

### Duration of Using Kavita

A year or so

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
There are no issues as long as you aren't using the Kavita name to sound official, like "Kavita mobile app", given Kavita plans to offer it's own. But usage of Kavita in the name is fine. 

So Kavita Import Tool is completely fine by me and if you reply with a link to the code, I can add it to the wiki and pin on the discord. I know many people really want that functionality.

**Reply by mackenly** [reply]
Awesome. I have some CD stuff to wrap up but will follow up soon.

**Reply by mackenly** [reply]
@majora2007 Following up on this. I've made it all [public](https://github.com/mackenly/kavita-import-tool), and it's ready as a soft "launch" but wanted to ask a quick question:
What is the preferred file formats for the various supported file types? Ex: I have `epub` as the preferred followed by `mobi` and `pdf`. Users can select their preferred type to override this. The functionality is that if a user selects multiple multiple files with the same name, only the preferred will be used. 

There isn't currently support for any of the complex file name parsing described in the [docs](https://wiki.kavitareader.com/guides/scanner/manga#examples-of-naming-conventions-supported-out-of-the-box), but that might could be added (I'm just not all that familiar with the use case, so stuck with ebooks for now).

Here's an example of the folder structure output:
```
kavita-import-tool-output-[epoch time]
  ┖── LearningAssemblyLanguage
        ┖── LearningAssemblyLanguage.epub 
      RustForTypeScriptDevelopers
          ┖── RustForTypeScriptDevelopers.pdf
      DesigningMultimediaPipelinesWithGStreamer
          ┖── DesigningMultimediaPipelinesWithGStreamer.mobi
```

**Reply by majora2007** [reply]
So Kavita doesn't support mobi (as it's a dead format with very limited library support to boot). In my personal preference epub then pdf as epub renders better, supports internal metadata, etc. 

That folder structure looks good to me.

**Reply by mackenly** [reply]
Ah, got it. I'm not sure where I noticed mobi. With comics and manga, are those pretty much always using cbz files?

**Reply by majora2007** [reply]
Not always. Here is the list of extensions we currently support:

```
public const string ImageFileExtensions = @"^(\.png|\.jpeg|\.jpg|\.webp|\.gif|\.avif)"; 
    public const string ArchiveFileExtensions = @"\.cbz|\.zip|\.rar|\.cbr|\.tar.gz|\.7zip|\.7z|\.cb7|\.cbt";
    public const string EpubFileExtension = @"\.epub";
    public const string PdfFileExtension = @"\.pdf";
```

**Reply by mackenly** [reply]
Awesome, I've updated the app to remove mobi and added support for all of those. Had a few deployment bugs to work out, but they should be squashed.

**Reply by majora2007** [reply]
I saw that. So are you ready for people to start looking at the tool? Would you like to announce it on our discord or should I?

**Reply by mackenly** [reply]
Yeah, you can if you'd like. I joined a little while ago @mackenly. 
Added another update a few moments ago to address some feedback I got, but other than that, I'm pretty happy with where it's at. Further things would be based on feedback.
